pubsub/streaming-analytics/README.md

```diff
@@ -0,0 +1,162 @@
+
+1. Setup the Cloud SDK to your GCP project.
+
+   ```bash
+   gcloud init
+   ```
+
+1. [Create a service account key] as a JSON file.
+   For more information, see [Creating and managing service accounts].
+
+   * From the **Service account** list, select **New service account**.
+   * In the **Service account name** field, enter a name.
+   * From the **Role** list, select **Project > Owner**.
+
+     > **Note**: The **Role** field authorizes your service account to access resources.
+     > You can view and change this field later by using the [GCP Console IAM page].
+     > If you are developing a production app, specify more granular permissions than **Project > Owner**.
+     > For more information, see [Granting roles to service accounts].
+
+   * Click **Create**. A JSON file that contains your key downloads to your computer.
+
+1. Set your `GOOGLE_APPLICATION_CREDENTIALS` environment variable to point to your service account key file.
+
+   ```bash
+   export GOOGLE_APPLICATION_CREDENTIALS=path/to/your/credentials.json
+   ```
+
+1. Create a Cloud Storage bucket.
+
+   ```bash
+   BUCKET_NAME=your-gcs-bucket
+   PROJECT_NAME=$(gcloud config get-value project)
+   
+   gsutil mb gs://$BUCKET_NAME
+   ```
+   
+ 1. Start a [Google Cloud Scheduler] job that publishes one message to a [Google Cloud Pub/Sub] topic every minute. This will create an [App Engine] app if one has never been created on the project. 
+ 
+    ```bash
+    # Create a Pub/Sub topic.
+    gcloud pubsub topics create cron-topic
+    
+    # Create a Cloud Scheduler job
+    gcloud scheduler jobs create pubsub publisher-job --schedule="* * * * *" \
+      --topic=cron-topic --message-body="Hello!"
+    
+    # Run the job. 
+    gcloud scheduler jobs run publisher-job
+    ```
+
+## Setup
+
+The following instructions will help you prepare your development environment.
+
+1. Download and install the [Java Development Kit (JDK)].
+   Verify that the [JAVA_HOME] environment variable is set and points to your JDK installation.
+
+1. Download and install [Apache Maven] by following the [Maven installation guide] for your specific operating system.
+
+1. Clone the `java-docs-samples` repository.
+
+    ```bash
+    git clone https://github.com/GoogleCloudPlatform/java-docs-samples.git
+    ```
+
+1. Navigate to the sample code directory.
+
+   ```bash
+   cd java-docs-samples/pubsub/streaming-analytics
+   ```
+
+## Streaming Analytics
+
+### Google Cloud Pub/Sub to Google Cloud Storage
+
+* [PubSubToGCS.java](src/main/java/com/examples/pubsub/streaming/PubSubToGCS.java)
+
+To run the example in Cloud Dataflow and write output files in the desired Cloud Storage location.
+
```

Comment:
Please add a short explanation of what the sample does. Something like:

The following example will run a streaming pipeline. It will read messages from a Pub/Sub topic, then window them into fixed-sized intervals, and write one file per window into a GCS location.

* `--project`: sets the Google Cloud project ID to run the pipeline on
* `--inputTopic`: sets the input Pub/Sub topic to read messages from
* `--output`: sets the output GCS path prefix to write files to
* `--runner` *[optional]*: specifies the runner to run the pipeline, defaults to "DirectRunner"
* `--windowSize` *[optional]*: specifies the window size in minutes, defaults to 1

Reply:
Good call! I improved the cleanup step to match runner selection.